onnxruntime/core/graph/contrib_ops/quantization_defs.cc

```diff
@@ -1246,5 +1247,233 @@ ONNX_MS_OPERATOR_SET_SCHEMA(
+            "This is also called adjs or adjustment in some frameworks.",
+            AttributeProto::INTS,
+            OPTIONAL_VALUE)
+        .Attr(
+            "dilations",
+            "dilation value along each spatial axis of the filter. If not present, the dilation defaults to 1 along each spatial axis.",
+            AttributeProto::INTS,
+            OPTIONAL_VALUE)
+        .Attr(
+            "strides",
+            "Stride along each spatial axis. If not present, the stride defaults to 1 along each spatial axis.",
+            AttributeProto::INTS,
+            OPTIONAL_VALUE)
+        .Attr("auto_pad",
+              "auto_pad must be either NOTSET, SAME_UPPER, SAME_LOWER or VALID. Where "
+              "default value is NOTSET",
+              AttributeProto::STRING, std::string("NOTSET"))
+        .Attr("pads", "Padding for the beginning and ending along each spatial axis", AttributeProto::INTS, OPTIONAL_VALUE)
+        .Attr(
+            "group",
+            "number of groups input channels and output channels are divided into.",
+            AttributeProto::INT,
+            static_cast<int64_t>(1))
+        .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+          auto x_type = ctx.getInputType(0);
+          auto w_type = ctx.getInputType(3);
+          if (nullptr == x_type || nullptr == w_type || x_type->value_case() != ONNX_NAMESPACE::TypeProto::kTensorType ||
+              w_type->value_case() != ONNX_NAMESPACE::TypeProto::kTensorType) {
+            fail_type_inference("inputs are expected to have tensor type.");
+          }
+
+          auto x_zero_point_type = ctx.getInputType(2);
+          if (nullptr == x_zero_point_type ||
+              x_zero_point_type->tensor_type().elem_type() != x_type->tensor_type().elem_type()) {
+            fail_type_inference("input and zero_point pair is expected to have be same type.");
+          }
+
+          auto w_zero_point_type = ctx.getInputType(5);
+          if (nullptr == w_zero_point_type ||
+              w_zero_point_type->tensor_type().elem_type() != w_type->tensor_type().elem_type()) {
+            fail_type_inference("weight and zero_point pair is expected to have same type.");
+          }
+          // To verify that we have valid input tensor
+          auto a_type = ctx.getInputType(0);
+          auto b_type = ctx.getInputType(3);
+
+          if (nullptr == a_type || nullptr == b_type || a_type->value_case() != ONNX_NAMESPACE::TypeProto::kTensorType ||
+              b_type->value_case() != ONNX_NAMESPACE::TypeProto::kTensorType) {
+            fail_type_inference("inputs are expected to have tensor type.");
+          }
+
+          // validate scale and zero points
+          // scale and zero points could be scalar or 1-D tensor which depends on quanization per-channel or per-tensor
+          ValidateTypeAndShapeForScaleAndZP(ctx, 1, ONNX_NAMESPACE::TensorProto::FLOAT, ScaleTensorType::Both);
```

Comment:
>ScaleTensorType::Both [](http://example.com/codeflow?start=88&length=21)

is it expected that scale of x supports per-channel?

Reply:
Thanks for pointing out the mistake. Only scale of kernel supports per-channel.